src/fides/api/ops/tasks/__init__.py

```diff
@@ -1,28 +1,40 @@
-from typing import Any, ContextManager, Dict, List, MutableMapping, Optional, Union
+from typing import Any, ContextManager, Dict, List, Optional
 
 from celery import Celery, Task
 from loguru import logger
 from sqlalchemy.orm import Session
-from toml import load as load_toml
 
 from fides.core.config import FidesConfig, get_config
-from fides.lib.db.session import get_db_session
+from fides.lib.db.session import get_db_engine, get_db_session
 
 CONFIG = get_config()
 MESSAGING_QUEUE_NAME = "fidesops.messaging"
 
 
 class DatabaseTask(Task):  # pylint: disable=W0223
-    _session = None
+    _task_engine = None
+    _sessionmaker = None
 
-    @property
-    def session(self) -> ContextManager[Session]:
-        """Creates Session once per process"""
-        if self._session is None:
-            SessionLocal = get_db_session(CONFIG)
-            self._session = SessionLocal()
+    def get_new_session(self) -> ContextManager[Session]:
+        """
+        Creates a new Session to be used for each task invocation.
 
-        return self._session
+        The new Sessions will reuse a shared `Engine` and `sessionmaker`
+        across invocations, so as to reuse db connection resources.
+        """
+        # only one engine will be instantiated in a given task scope, i.e
+        # once per celery process.
+        if self._task_engine is None:
+            _task_engine = get_db_engine(config=CONFIG)
+
+        # same for the sessionmaker
+        if self._sessionmaker is None:
+            self._sessionmaker = get_db_session(config=CONFIG, engine=_task_engine)
```

Comment:
Now that two sessions can concurrently change the database's state we should consider changing the `autocommit` and `autoflush` defaults to `True`. Currently in the execution model a privacy request's execution is locked to one worker, but the app server and a worker could theoretically be editing the same privacy request concurrently.

Reply:
@adamsachs and I discussed this synchronously! This can be addressed separately.

Reply:
per @seanpreston's slack
>  I don’t think it’s an immediate issue given it’s unlikely for the app server and worker to be saving the same privacy request at the same time

we will create a follow up to look into this further (GH issue coming soon)

Reply:
created https://github.com/ethyca/fides/issues/2508 for follow up